File: nysol_python/mcmd/methods/convTex/db/smp/func_argsize.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='argsize'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,v1,v2,v3
1,1,2,3
2,-5,2,1
3,1,,3
4,,,
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``"v"`` で始まる項目名の数を数える。
'''
script['sh_code']='''
mcal c='argsize($s{v*})' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='argsize($s{v*})', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/jp/examples/func_argsize.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,v1,v2,v3
1,1,2,3
2,-5,2,1
3,1,,3
4,,,
EOF
)}

############## 例1
title="基本例"
comment=<<'EOF'
\verb|"v"|で始まる項目名の数を数える。
EOF
scp=<<'EOF'
more dat1.csv
mcal c='argsize($s{v*})' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

